autogen.sh

```diff
@@ -316,39 +254,119 @@ EOF
+    cd $SRCROOTDIR/$target_dir
+    for i in $SRCROOTDIR/$patch_path/*.patch; do
+        patch -p1 -N --dry-run --silent < $i 2>/dev/null
+        if [ $? -eq 0 ]; then
+            patch -p1 -N < $i
+        fi
+    done
+    cd -
+}
+
+check_submodule_presence src/hwloc
+apply_patches maint/patches/pre/hwloc src/hwloc
+
+########################################################################
 ## Set up external packages
 ########################################################################
 
 # external packages that require autogen.sh to be run for each of them
 externals="src/pm/hydra src/mpi/romio src/openpa src/hwloc"
 
 if [ "yes" = "$do_izem" ] ; then
+    check_submodule_presence src/izem
     externals="${externals} src/izem"
 fi
 
 if [ "yes" = "$do_ucx" ] ; then
+    check_submodule_presence src/mpid/ch4/netmod/ucx/ucx
     externals="${externals} src/mpid/ch4/netmod/ucx/ucx"
 fi
 
 if [ "yes" = "$do_ofi" ] ; then
+    check_submodule_presence src/mpid/ch4/netmod/ofi/libfabric
     externals="${externals} src/mpid/ch4/netmod/ofi/libfabric"
 fi
 
 ########################################################################
+# This used to be an optionally installed hook to help with git-svn
+# versions of the old SVN repo.  Now that we are using git, this is our
+# mechanism that replaces relative svn:externals paths, such as for
+# "confdb" and "mpl". The basic plan is to delete the destdir and then
+# copy all of the files, warts and all, from the source directory to the
+# destination directory.
 echo
-echo "##########################################"
-echo "## Patching necessary third-party software"
-echo "##########################################"
+echo "####################################"
+echo "## Replicating confdb (and similar)"
+echo "####################################"
 echo
 
-# ucx
-if [ "yes" = "$do_ucx" ] ; then
-( cd src/mpid/ch4/netmod/ucx/ucx && for i in ../../../../../../maint/patches/pre/ucx/*.patch; do patch -p1 < $i; done )
-fi
+sync_external () {
+    srcdir=$1
+    destdir=$2
+
+    echo "syncing '$srcdir' --> '$destdir'"
+
+    # deletion prevents creating 'confdb/confdb' situation, also cleans
+    # stray files that may have crept in somehow
+    rm -rf "$destdir"
+    cp -pPR "$srcdir" "$destdir"
+}
 
-# hwloc
-( cd src/hwloc && for i in  ../../maint/patches/pre/hwloc/*.patch; do patch -p1 < $i; done )
+confdb_dirs=
+confdb_dirs="${confdb_dirs} src/mpi/romio/confdb"
+confdb_dirs="${confdb_dirs} src/mpi/romio/mpl/confdb"
+confdb_dirs="${confdb_dirs} src/mpl/confdb"
+confdb_dirs="${confdb_dirs} src/pm/hydra/confdb"
+confdb_dirs="${confdb_dirs} src/pm/hydra/mpl/confdb"
+confdb_dirs="${confdb_dirs} test/mpi/confdb"
+
+# hydra's copies of mpl and hwloc
+sync_external src/mpl src/pm/hydra/mpl
+
+# ROMIO's copy of mpl
+sync_external src/mpl src/mpi/romio/mpl
+
+# all the confdb directories, by various names
+for destdir in $confdb_dirs ; do
+    sync_external confdb "$destdir"
+done
```

Comment:
It's weird to have actual copies be done before the environment checks are complete (e.g., autotools check).